src/BaseRoutes.jsx

```diff
@@ -26,20 +20,19 @@ import debug from './utils/debug'
  * @return {Object}
  */
 export default function BaseRoutes({testElt = null}) {
-  const location = useLocation()
-  const navigate = useNavigate()
+  const location = useRef(useLocation())
+  const navigation = useRef(useNavigate())
```

Comment:
When you call `useEffect`, you are expected to declare your dependencies: the things that should trigger when your side effect is executed. This invocation of `useEffect` relies on the values of the history and location objects provided by React Router.

However, if one lists `useNavigate` as a direct dependency, then you end up with a render loop when the underlying React code attempts to compare the value of all dependencies pre- and post-render. Keep in mind, that if any of the side effect's dependencies is a function (such as `useNavigate`), it will execute it and store its result for future comparisons.

The ultimate result is the initial render calls your side-effect hook, then React evaluates your dependencies which results in a call to `useNavigate()`, which under the hood triggers a re-render, which then triggers your side-effect hook, so on and so forth.

To avoid this, we use `useRef` to store a _reference_ (or one can call it a pointer) to the history and navigation objects, and list the references as our dependencies. With this method, React is comparing the values returned by the React Router hooks rather than entering the infinite execution loop.

Reply:
oh that makes sense!